Orange/widgets/classify/owadaboost.py

```diff
@@ -54,13 +61,18 @@ def create_learner(self):
             n_estimators=self.n_estimators,
             learning_rate=self.learning_rate,
             preprocessors=self.preprocessors,
-            algorithm=self.losses[self.algorithm]
-        )
+            algorithm=self.losses[self.algorithm])
 
     def set_base_learner(self, learner):
-        self.base_estimator = learner if learner \
-            else self.DEFAULT_BASE_ESTIMATOR
-        self.base_label.setText("Base estimator: " + self.base_estimator.name)
+        self.Error.no_weight_support.clear()
+        if learner and not learner.supports_weights:
+            # Clear the error and reset to default base learner
+            self.Error.no_weight_support()
+            self.base_estimator = self.DEFAULT_BASE_ESTIMATOR
+        else:
+            self.base_estimator = learner or self.DEFAULT_BASE_ESTIMATOR
+            self.base_label.setText(
```

Comment:
This should be outside of the else block (remove one indent).
It sets the base_estimator correctly now, but not the label.